module_utils/certificate/providers/certmonger.py

```diff
@@ -141,5 +141,11 @@ def request_certificate(self):
             # Don't attempt to renew when near to expiration
             command += ["-R"]
 
+        # Set Kerberos principal
+        for principal in self.csr.principal:
+            command += ["-K", principal]
```

Comment:
do you need to `shlex.quote` this string?

Reply:
That's a good question. It might be safer adding it not only here but in other places as well (line `common_name`.

Reply:
Using `shlex.quote` is actually adding undesired quotes to the result. For example:
```
"CN=My Certificate with SAN != CN='My Certificate with SAN'"
```
I'm calling `run_command` which will call `Popen` using the list as the command. It seems quoting is not required when using `Popen` this way. It would be required in case I was using a `str` command and `shell=True`.

Reply:
ok - as long as it is shell safe